src/test/run-pass/extern-types-size_of_val.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(extern_types)]
+
+use std::mem::{size_of_val, align_of_val};
+
+extern {
+    type A;
+}
+
+fn main() {
+    let x: &A = unsafe {
+        &*(1usize as *const A)
+    };
+
+    assert_eq!(size_of_val(x), 0);
+    assert_eq!(align_of_val(x), 1);
+}
```

Comment:
More tests I think we need:

1) test that pointers to extern types have the same *size* and *align* as a thin pointer.
2) check that pointers to extern types have the right ABI - maybe add a test to https://github.com/rust-lang/rust/tree/e22a3cf7e12b622b143694942504c66a544dc4c5/src/test/run-make/extern-fn-struct-passing-abi
3) check that structs with `extern` tails work - e.g. `struct Foo<T: ?Sized>(u8, T);` where `T = <foreign type>/Foo<<foreign type>>`